src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IUsingStatement.vb

```diff
@@ -192,33 +262,672 @@ Module Program
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC36010: 'Using' operand of type 'Program.C' must implement 'System.IDisposable'.
+        Using c1 As New C'BIND:"Using c1 As New C"
+              ~~~~~~~~~~~
+]]>.Value
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of UsingBlockSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IUsingStatement_InvalidEmptyUsingResource()
+            Dim source = <![CDATA[
+Option Strict On
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Using'BIND:"Using"
+        End Using
+    End Sub
+End Module]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUsingStatement (OperationKind.UsingStatement, IsInvalid) (Syntax: 'Using'BIND: ... End Using')
+  Resources: 
+    IInvalidExpression (OperationKind.InvalidExpression, Type: null, IsInvalid) (Syntax: '')
+      Children(0)
+  Body: 
+    IBlockStatement (0 statements) (OperationKind.BlockStatement, IsInvalid) (Syntax: 'Using'BIND: ... End Using')
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC30201: Expression expected.
+        Using'BIND:"Using"
+             ~
+]]>.Value
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of UsingBlockSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IUsingStatement_InvalidNothingResources()
+            Dim source = <![CDATA[
+Option Strict On
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Using Nothing'BIND:"Using Nothing"
+        End Using
+    End Sub
+End Module]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUsingStatement (OperationKind.UsingStatement, IsInvalid) (Syntax: 'Using Nothi ... End Using')
+  Resources: 
+    IConversionExpression (Implicit, TryCast: False, Unchecked) (OperationKind.ConversionExpression, Type: System.Object, Constant: null, IsInvalid, IsImplicit) (Syntax: 'Nothing')
+      Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+      Operand: 
+        ILiteralExpression (OperationKind.LiteralExpression, Type: null, Constant: null, IsInvalid) (Syntax: 'Nothing')
+  Body: 
+    IBlockStatement (0 statements) (OperationKind.BlockStatement, IsInvalid) (Syntax: 'Using Nothi ... End Using')
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC36010: 'Using' operand of type 'Object' must implement 'System.IDisposable'.
+        Using Nothing'BIND:"Using Nothing"
+              ~~~~~~~
+]]>.Value
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of UsingBlockSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+        <CompilerTrait(CompilerFeature.IOperation)>
```

Comment:
><CompilerTrait(CompilerFeature.IOperation)> [](start = 8, length = 43)

Consider adding an empty line between this and the previous function.

Reply:
Added.

---
In reply to: [144355957](https://github.com/dotnet/roslyn/pull/22630#discussion_r144355957) [](ancestors = 144355957)